src/toil/job.py

```diff
@@ -798,10 +848,12 @@ def effectiveRequirements(self, config):
         :rtype: Expando
         :return: a dictionary/object hybrid with one entry/attribute for each requirement
         """
+        if self._config is None:
```

Comment:
I thought about eliminating this but effectiveRequirements is still useful as a way to wrap up resource requirements into a dictionary/expando. What do you think? IMO, it's pretty convenient especially for argument unpacking so I'm in favor of keeping it.